lib/ts/recipe/emailpassword/components/features/emailVerification/emailVerification.tsx

```diff
@@ -0,0 +1,296 @@
+                    .getAppInfo()
+                    .websiteBasePath.getAsStringDangerous(),
+                undefined,
+                this.props.history
+            );
+        } catch (e) {}
+        return;
+    };
+
+    getSessionRecipe(): Session | undefined {
+        return SuperTokens.getDefaultSessionRecipe();
+    }
+
+    doesSessionExist = async (): Promise<boolean> => {
+        // If props provided by user.
+        if (this.props.doesSessionExist !== undefined) {
+            return await this.props.doesSessionExist();
+        }
+
+        const sessionRecipe = this.getSessionRecipe();
+        if (sessionRecipe !== undefined) {
+            return sessionRecipe.doesSessionExist();
+        }
+
+        // Otherwise, return false.
+        return false;
+    };
+
+    redirectToVerifyEmailScreen = async (): Promise<void> => {
+        redirectToInApp(
+            `${this.getRecipeInstanceOrThrow()
+                .getAppInfo()
+                .websiteBasePath.getAsStringDangerous()}${DEFAULT_VERIFY_EMAIL_PATH}?rid=emailpassword`,
+            undefined,
+            undefined
+        ); // No history object provided here, we need to reload the page.
+        return;
+    };
+
+    onSuccessfulEmailVerificationContinueClicked = async (): Promise<void> => {
+        redirectToInApp(
+            this.getRecipeInstanceOrThrow().getConfig().signInAndUpFeature.onSuccessRedirectURL,
+            undefined,
+            this.props.history
+        );
+        return;
+    };
+
+    async componentDidMount(): Promise<void> {
+        const hasToken = this.state.token.length !== 0;
+
+        // Redirect to login if no existing session and no token in URL.
+        const hasValidSession = await this.doesSessionExist();
+        if (hasValidSession === false && hasToken === false) {
+            redirectToInApp(
+                this.getRecipeInstanceOrThrow()
+                    .getAppInfo()
+                    .websiteBasePath.getAsStringDangerous(),
+                undefined,
+                this.props.history
+            );
+        }
+
+        try {
+            if (hasToken === false) {
+                const isEmailAlreadyVerified = await this.getRecipeInstanceOrThrow().isEmailVerified();
+                if (isEmailAlreadyVerified === true) {
+                    redirectToInApp(
+                        this.getRecipeInstanceOrThrow().getConfig().signInAndUpFeature.onSuccessRedirectURL,
+                        undefined,
+                        this.props.history
+                    );
+                }
+            }
+        } catch (e) {}
+
+        return;
+    }
+
+    render = (): JSX.Element => {
+        // In case Email Verification Mode is not required, redirect to onSuccessRedirectURL.
+        if (
+            this.getRecipeInstanceOrThrow().getConfig().emailVerificationFeature.mode !==
+            EMAIL_VERIFICATION_MODE.REQUIRED
+        ) {
+            redirectToInApp(
```

Comment:
I don't think doing a redirect inside a render function is appropriate. Perhaps this should be done in componentDidMount